compiler/noirc_evaluator/src/brillig/brillig_ir/entry_point.rs

```diff
@@ -23,16 +23,11 @@ impl<F: AcirField + DebugToString> BrilligContext<F, Stack> {
         return_parameters: Vec<BrilligParameter>,
         target_function: FunctionId,
         globals_init: bool,
-        mut globals_memory_size: usize,
+        globals_memory_size: usize,
         options: &BrilligOptions,
     ) -> BrilligArtifact<F> {
         let mut context = BrilligContext::new(options);
 
-        if options.enable_array_copy_counter {
-            // Reserve space for the array clone counter
-            globals_memory_size += 1;
```

Comment:
We never needed this as we are allocating during `compile_globals`. As this is in `new_entry_point_artifact` (which uses a fresh context) as well, this logic would simply offset where we start the calldata offset by an additional slot that was never allocated. 

Something like this would have been needed if we never allocated for the debug slot, but we are and I think it is cleaner that way.